### PR TITLE
Replace node-fetch with cross-fetch in r11s-driver

### PR DESCRIPTION
**Pull request by znewton**
Discussion in #8725 brought to light that node-fetch is not working in the browser when using bundlers that do not automatically include Node polyfills (Webpack 5, Vite, etc). Switching to `cross-fetch` is easy enough.

**cross-fetch vs isomorphic-fetch**
[cross-fetch](https://www.npmjs.com/package/cross-fetch)
-  ~11 million weekly downloads
- Last published 2 months ago
- [8kB (2.8kB gzipped)](https://bundlephobia.com/package/cross-fetch@3.1.5) bundle size

[isomorphic-fetch](https://www.npmjs.com/package/isomorphic-fetch)
- ~7 million weekly downloads
- Last published 1 year ago
- [9.2kB (3.1kB gzipped)](https://bundlephobia.com/package/isomorphic-fetch@3.0.0) bundle size

### CODE REVIEW

File: packages/drivers/routerlicious-driver/src/test/restWrapper.spec.ts [hunk 1]
```diff
@@ -7,67 +7,49 @@ import assert from "assert";
 import { TelemetryUTLogger } from "@fluidframework/telemetry-utils";
 import { DriverErrorType } from "@fluidframework/driver-definitions";
 import { RateLimiter } from "@fluidframework/driver-utils";
-import fetchMock from "fetch-mock";
-import * as nodeFetchModule from "node-fetch";
-import { reset, replace } from "sinon";
 import { RouterliciousRestWrapper } from "../restWrapper";
 import { R11sErrorType } from "../errorUtils";
+// eslint-disable-next-line @typescript-eslint/no-require-imports,import/order
```
Comment (tylerbutler) [inline packages/drivers/routerlicious-driver/src/test/restWrapper.spec.ts:12]: nit: do you have to disable import/order?

File: packages/drivers/routerlicious-driver/src/test/restWrapper.spec.ts [hunk 2]
```diff
@@ -77,46 +59,49 @@ describe("RouterliciousDriverRestWrapper", () => {
         await restWrapper.load();
     });
     after(() => {
-        reset();
+        // reset();
+        nock.restore();
```
Comment (tylerbutler) [inline packages/drivers/routerlicious-driver/src/test/restWrapper.spec.ts:63]: nit

```suggestion
        nock.restore();
```

**Review by tylerbutler — APPROVED** [review]
Thanks for doing this, @znewton! I didn't look closely at the code, just the dependencies and lint stuff.

File: packages/drivers/routerlicious-driver/src/test/restWrapper.spec.ts (hunk 1 above)
Comment (znewton) [reply packages/drivers/routerlicious-driver/src/test/restWrapper.spec.ts]: Ope, yeah my VSCode wasn't recognizing that esModuleInterop: true was set. Switched this to a normal import and put it in order.

**Pull request merged**